src/login/LoginOptionsScreen.tsx

```diff
@@ -0,0 +1,230 @@
+import {AfterLoginParams} from '@atb/login/types';
+import {VippsLoginButton} from '@atb/components/vipps-login-button';
+import MessageBox from '@atb/components/message-box';
+import * as Sections from '@atb/components/sections';
+import {APP_SCHEMA} from '@env';
+
+const themeColor: StaticColorByType<'background'> = 'background_accent_0';
+
+export type LoginOptionsRouteParams = {
+  afterLogin: AfterLoginParams;
+};
+
+type LoginOptionsRouteProps = RouteProp<
+  LoginInAppStackParams,
+  'LoginOptionsScreen'
+>;
+
+type LoginOptionsProps = {
+  route: LoginOptionsRouteProps;
+};
+
+export default function LoginOptionsScreen({
+  route: {
+    params: {afterLogin},
+  },
+}: LoginOptionsProps) {
+  const {t} = useTranslation();
+  const styles = useThemeStyles();
+  const {signInWithCustomToken} = useAuthState();
+  const [error, setError] = useState<VippsSignInErrorCode>();
+  const [isLoading, setIsLoading] = useState(false);
+  const navigation = useNavigation();
+  const appStatus = useAppStateStatus();
+  const [authorizationCode, setAuthorizationCode] = useState<
+    string | undefined
+  >(undefined);
+
+  const authenticateUserByVipps = async () => {
+    setIsLoading(true);
+    try {
+      await authorizeUser(setIsLoading);
+    } catch (err) {
+      setError('unknown_error');
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(
+    () =>
+      navigation.addListener('focus', () => {
+        setIsLoading(false);
+      }),
+    [navigation],
+  );
+
+  const signInUsingCustomToken = async (token: string) => {
+    const response = await signInWithCustomToken(token);
+    if (!response.error) {
+      navigation.navigate(afterLogin.routeName as any, afterLogin.routeParams);
+    } else {
+      setError(response.error);
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    const signInVippsUser = async (authorizationCode: string) => {
+      try {
+        const customToken = await getOrCreateVippsUserCustomToken(
+          authorizationCode,
+        );
+        await signInUsingCustomToken(customToken.data as string);
+        await storage.set('vipps_state', '');
+        await storage.set('vipps_nonce', '');
+      } catch (err) {
+        setError('unknown_error');
+        setIsLoading(false);
+      }
+    };
+    if (authorizationCode) {
+      signInVippsUser(authorizationCode);
+    }
+  }, [authorizationCode]);
+
+  useEffect(() => {
+    const vippsCallbackHandler = async (event: any) => {
+      if (event.url.includes(`${APP_SCHEMA}://auth/vipps`)) {
+        setIsLoading(true);
+        const code = parseUrl(event.url).query.code;
+        if (code) {
+          const state = parseUrl(event.url).query.state;
+          const initialState = await storage.get('vipps_state');
+          if (initialState?.toString() !== state?.toString()) {
+            setError('unknown_error');
+            setIsLoading(false);
+          }
+          setAuthorizationCode(code.toString());
```

Comment:
This should be called and changing the code even if `initialState?.toString() !== state?.toString()`? No consequences for when calling `getOrCreateVippsUserCustomToken` above?

Reply:
You are so right! This should not be called. I started with throwing the error when the state dint match so it should have stopped the flow. But now when we are just setting the state, this should be done in else block